src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1286,14 +1297,37 @@ public Worker build() {
                 workerStateChangeListener = DEFAULT_WORKER_STATE_CHANGE_LISTENER;
             }
 
-            if(leaseCleanupValidator == null) {
-                leaseCleanupValidator = DEFAULT_LEASE_CLEANUP_VALIDATOR;
+            if ((shardSyncer == null && leaseCleanupValidator != null) || (shardSyncer != null && leaseCleanupValidator == null)) {
```

Comment:
If you have access to apache commons, replace this with ObjectUtils#allNotNull for readability.

Reply:
Discussed offline. Leaving as-is for now.